pkg/agent/systemdmanager.go

```diff
@@ -143,14 +148,14 @@ func (m *SystemdManager) Run(ctx context.Context) error {
 }
 
 func (m *SystemdManager) reconcileUnit(ctx context.Context, unit string) error {
-	f, err := os.Open(fmt.Sprintf("/sys/fs/cgroup/systemd/system.slice/%s/cgroup.procs", unit))
+	f, err := os.Open(fmt.Sprintf("%s/%s/cgroup.procs", m.cgroupPath, unit))
 	if os.IsNotExist(err) {
 		m.mtx.Lock()
 		p := m.unitProfilers[unit]
 		if p != nil {
 			p.Stop()
 		}
```

Comment:
```suggestion
		if p, ok := m.unitProfilers[unit]; ok && p != nil {
			p.Stop()
		}
```

Just a nitpicking. With the below change this would be more accurate.